### ISSUE TITLE
Inconsistent results on comparing calculated probability with topic label

### ISSUE THREAD

**Issue by raghavmallampalli** [MaartenGr/BERTopic#185]
I have trained the model on a small training set (~3500) samples. I find that the maximum probability solution is not the same as the topic that has been assigned.

For example, in the following picture, where I have calculated the 3 most suitable labels for the training set from the probabilities returned by the model using `.fit_transform()`, the second document (index 1) is given topic 4 by probability and topic 37 by the model. 
![image](https://user-images.githubusercontent.com/49435792/126434741-4d7c6702-f577-4a71-854e-85a53e6ac913.png)

Am I wrong in assuming that the assigned label is the one with maximum probability?

Additionally, I find that the clustering performs poorly while transforming unseen data. All the documents are assigned to the same 2-3 labels or to noise. Is there anything I should be watching out for while training so the model can generalise better?

**Comment by MaartenGr** [comment]
Thank you for the extensive description. The inconsistent results between probabilities is a known issue and a fix for that can already be found [here](https://github.com/MaartenGr/BERTopic/pull/176). 

With respect to the clustering, this might be due to the small dataset that you are currently using. BERTopic performs especially well on higher datasets unless you update some of the hyperparameters, such as `min_topic_size` to increase the number of topics generated. However, that will result in less accurate topic representations as the c-TF-IDF method has too little data.

**Comment by raghavmallampalli** [comment]
1. Hi, thanks a lot for the quick reply! I will try the patches and close the issue ASAP.
 
2. I will try obtaining more data as you have pointed out. I don't think reducing the minimum topic size is a viable option for me, but thanks for the suggestion!

**Comment by raghavmallampalli** [comment]
The fix works! If there are others who urgently need the fix, you can install the updated version with (make sure to remove the previously installed version first):
`!pip install git+git://github.com/MaartenGr/BERTopic.git@refs/pull/176/head`
